src/sentry_core.c

```diff
@@ -448,18 +448,40 @@ sentry__capture_event(sentry_value_t event)
 }
 
 bool
-sentry__should_send_transaction(sentry_value_t tx_cxt)
+sentry__should_send_transaction(
+    sentry_value_t tx_ctx, sentry_sampling_context_t *sampling_ctx)
 {
-    sentry_value_t context_setting = sentry_value_get_by_key(tx_cxt, "sampled");
-    if (!sentry_value_is_null(context_setting)) {
-        return sentry_value_is_true(context_setting);
-    }
+    sentry_value_t context_setting = sentry_value_get_by_key(tx_ctx, "sampled");
+    bool sampled = sentry_value_is_null(context_setting)
+        ? false
+        : sentry_value_is_true(context_setting);
+    sampling_ctx->parent_sampled
+        = sentry_value_is_null(context_setting) ? NULL : &sampled;
 
+    int parent_sampled_int; // bool to int conversion
```

Comment:
`nit`: How about being more explicit here? I see the value is only read if has been previously set, so it should be fine. But in case someone makes changes to the code below in future, this could easily slip through.
```suggestion
    int parent_sampled_int = 0; // bool to int conversion
```